packages/core/useCountdown/index.ts

```diff
@@ -0,0 +1,98 @@
+import { toValue, useIntervalFn } from '@vueuse/shared'
+import type { MaybeRefOrGetter, Pausable } from '@vueuse/shared'
+import type { Ref } from 'vue-demi'
+import { ref } from 'vue-demi'
+
+export interface UseCountdownOptions {
+  /**
+   *  Interval for the countdown in milliseconds. Default is 1000ms.
+   */
+  interval?: MaybeRefOrGetter<number>
+  /**
+   * Callback function called when the countdown reaches 0.
+   */
+  onComplete?: () => void
+  /**
+   * Callback function called on each tick of the countdown.
+   */
+  onTick?: () => void
+  /**
+   * Start the countdown immediately
+   *
+   * @default false
+   */
+  immediate?: boolean
+}
+
+export interface UseCountdownReturn extends Pausable {
+  /**
+   * Current countdown value.
+   */
+  remaining: Ref<number>
+  /**
+   * Resets the countdown and repeatsLeft to their initial values.
+   */
+  reset: () => void
+  /**
+   * Stops the countdown and resets its state.
+   */
+  stop: () => void
+  /**
+   * Reset the countdown and start it again.
+   */
+  start: (initialCountdown?: MaybeRefOrGetter<number>) => void
+}
+
+/**
+ * Wrapper for `useIntervalFn` that provides a countdown timer.
+ *
+ * @param initialCountdown
```

Comment:
We should specify that the countdown is in second to avoid confusing users since the interval is in ms